### DISCUSSION TITLE
x-init fires multiple times

### DISCUSSION THREAD

**Discussion by pjdevries** [alpinejs/alpine#1732] [7. Bugs (archived)]
I have a situation in which `init` function of my data component is executed multiple times. I am using AlpineJS 3.2.1. Is this a known issue? Are there known circumstances in which it occurs? If so, I love to hear if there is a solution or workaround.

**Comment by danddanddand** [comment]
It would help to have code example, but one possible gotcha is if you have x-init="init()" it would be redundant since Alpine will check for an init function's existence and call it automatically.

**Reply by pjdevries** [reply]
Thanx @danddanddand. I have an `init()` in my `x-data` component and no `x-init`. So not an issue there. I did try with a `setup()` in my component and `x-init="setup()"`, but that does not solve the problem.

My code is not the most professional and a bit application specific. So before posting it here, I'll do some more testing and cleaning first :)

**Comment by davidwebca** [comment]
I'll be opening an issue, as I was searching for an answer, I found out your post and realized it's happening only when we call init() manually inside x-init. As of Alpine v3, init is called automatically if it's found on a component, but it seems it doesn't check the expression before and doesn't allow us to use x-init to call a function named "init", not even to pass $watch and other magics.

```html
<!-- Does not work, $watch is not passed and init is called twice -->
<div x-init="init($watch)"></div>
```

**Reply by davidwebca** [reply]
Me going there trying to open an issue when I suddenly realize they swtiched the templates to not allow opening issues. I facepalmed reallll hard lmao.
